pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/OneOf.java

```diff
@@ -47,7 +47,7 @@ public boolean equals(Object o) {
 
     @Override
     public int hashCode() {
-        return Objects.hash(kind, value);
+        return kind.hashCode() * 31 + Objects.hashCode(value);
```

Comment:
Why not essentially copy `Arrays.hashCode()` and preserve the value?
```
return (31 + kind.hashCode()) * 31 + (value == null ? 0 : value.hashCode());
```

Reply:
Not sure why this might be important, but okay, let's preserve the value.

Reply:
BTW, I confirm that I've verified that the new value matches the old implementation. However, I'm still rewriting the unit test so that it tests our implementation rather than assumes that it matches Objects.hash().

Reply:
`MerkleDb` uses `key.hashCode()` internally. It it ever changes we have to do data migration.